Handle communication from main app

jasonzurita opened the pull request:
This will allow the apple watch to process feeding actions. Still need to have the watch _sync up_ with the main app when coming to life for the first time, but this is already a good improvement.

jasonzurita committed:
feat: preform watch comms first then local state

This better address the _async_ nature of the
watch communication. We can now handle failure
communication cases better.

This also:
- Removes the messy higher order feeding communication reducer
- Sets up for receiving feeding state changes from the main app!

In the future, having the watch communication logic move out of the
individual views would be nice, but sending async actions is difficult
to handle right now.

FlippingLeaf commented on BabyPatterns/DispatchVC.swift:
```diff
@@ -199,7 +199,11 @@ extension DispatchVC: WCSessionDelegate {
     }
 
-    func session(_ session: WCSession, didReceiveMessageData messageData: Data) {
+    func session(_ session: WCSession, didReceiveMessageData messageData: Data, replyHandler: @escaping (Data) -> Void) {
```
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/jasonzurita/BabyPatterns/blob/16cbfbb746a30284804adc990b71a4559002884e/BabyPatterns/DispatchVC.swift#L202"></span>Line should be 120 characters or less: currently 121 characters<br />
<code>line_length</code> <code>DispatchVC.swift:202</code></td>
    </tr>
  </tbody>
</table>

FlippingLeaf commented on BabyPatterns/FeedingsVM+FeedingInProgress.swift:
```diff
@@ -91,4 +105,23 @@ extension FeedingsVM {
+
+        WCSession.default.sendMessageData(d, replyHandler: { _ in
+            // no op
+        }) { error in
```
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/jasonzurita/BabyPatterns/blob/16cbfbb746a30284804adc990b71a4559002884e/BabyPatterns/FeedingsVM+FeedingInProgress.swift#L122"></span>Trailing closure syntax should not be used when passing more than one closure argument.<br />
<code>multiple_closures_with_trailing_closure</code> <code>FeedingsVM+FeedingInProgress.swift:122</code></td>
    </tr>
  </tbody>
</table>

FlippingLeaf commented on Watch Extension/AddFeedingView.swift:
```diff
@@ -75,21 +76,45 @@ struct AddFeedingView: View {
                         .gesture(TapGesture().onEnded {
-                            self.store.send(.feeding(.start(type: self.feedingIntent, side: .left)))
-                            self.isShowingSheet = false
+                            
```
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/jasonzurita/BabyPatterns/blob/16cbfbb746a30284804adc990b71a4559002884e/Watch Extension/AddFeedingView.swift#L79"></span>Lines should not have trailing whitespace.<br />
<code>trailing_whitespace</code> <code>AddFeedingView.swift:79</code></td>
    </tr>
  </tbody>
</table>

FlippingLeaf commented on Watch Extension/AddFeedingView.swift:
```diff
@@ -75,21 +76,45 @@ struct AddFeedingView: View {
+        WCSession.default.sendMessageData(d, replyHandler: { _ in
+            self.store.send(.feeding(.start(type: type, side: side)))
+            self.isShowingSheet = false
+        }) { error in
```
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/jasonzurita/BabyPatterns/blob/16cbfbb746a30284804adc990b71a4559002884e/Watch Extension/AddFeedingView.swift#L112"></span>Trailing closure syntax should not be used when passing more than one closure argument.<br />
<code>multiple_closures_with_trailing_closure</code> <code>AddFeedingView.swift:112</code></td>
    </tr>
  </tbody>
</table>

FlippingLeaf commented on Watch Extension/FeedingView.swift:
```diff
@@ -16,15 +18,43 @@ struct FeedingView: View {
+            case .resume:
+                self.store.send(.feeding(.resume(type: type)))
+            }
+        }) { error in
```
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/jasonzurita/BabyPatterns/blob/16cbfbb746a30284804adc990b71a4559002884e/Watch Extension/FeedingView.swift#L51"></span>Trailing closure syntax should not be used when passing more than one closure argument.<br />
<code>multiple_closures_with_trailing_closure</code> <code>FeedingView.swift:51</code></td>
    </tr>
  </tbody>
</table>

FlippingLeaf commented on Watch Extension/SessionCoordinator.swift:
```diff
@@ -33,4 +34,28 @@ extension SessionCoordinator: WCSessionDelegate {
         }
     }
+
+    func session(_ session: WCSession, didReceiveMessageData messageData: Data, replyHandler: @escaping (Data) -> Void) {
```
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/jasonzurita/BabyPatterns/blob/16cbfbb746a30284804adc990b71a4559002884e/Watch Extension/SessionCoordinator.swift#L38"></span>Line should be 120 characters or less: currently 121 characters<br />
<code>line_length</code> <code>SessionCoordinator.swift:38</code></td>
    </tr>
  </tbody>
</table>

jasonzurita committed:
feat: reflect actions from main app on apple watch

This will allow the apple watch to process
feeding actions. Still need to have the watch
_sync up_ with the main app when coming to life
for the first time, but this is already a good improvement.
